### PR TITLE
compose inventory_hostname

### LINKED ISSUE #1070: Option to import hosts with hostname/shortname only
##### SUMMARY
A option which we could use in the plugin to remove the domain part from "host.domain" when importing a Foreman inventory would be very useful to have a inventory only with hostname/shortname of hosts imported from Foreman.

##### ISSUE TYPE
 - Feature Idea

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: #1070

### CODE REVIEW

File: plugins/inventory/foreman.py [hunk 1]
```diff
@@ -492,87 +539,93 @@ def _populate_report_api(self):
...
             hostvars = self.inventory.get_host(host_name).get_vars()
             self._set_composite_vars(self.get_option('compose'), hostvars, host_name, strict)
             self._add_host_to_composed_groups(self.get_option('groups'), hostvars, host_name, strict)
             self._add_host_to_keyed_groups(self.get_option('keyed_groups'), hostvars, host_name, strict)
 
     def _populate_host_api(self):
+        hostnames = self.get_option('hostnames')
+        strict = self.get_option('strict')
         for host in self._get_hosts():
-            if host.get('name'):
```
Comment (evgeni) [inline plugins/inventory/foreman.py:503]: while this looks like a big change in the diff, it's actually just mostly re-indenting the code

**Comment by evgeni** [comment]
this is inspired by the [vmware](https://github.com/ansible-collections/community.vmware/blob/main/plugins/inventory/vmware_vm_inventory.py) inventory, which supports similar composition

File: plugins/inventory/foreman.py (hunk 1 above)
Comment (nesanton) [reply plugins/inventory/foreman.py:503]: took me a while to comprehend that ;)

**Comment by nesanton** [comment]
fwiw lgtm

File: plugins/inventory/foreman.py [hunk 2]
```diff
@@ -144,16 +144,29 @@
...
+password: changeme
 host_filters: 'organization="Web Engineering"'
+
+# shortname.foreman.yml
+plugin: theforeman.foreman.foreman
+url: https://foreman.example.com
+user: ansibleinventory
+password: changeme
+hostnames:
+  - name.split('.')[0]
```
Comment (nesanton) [inline plugins/inventory/foreman.py:169]: Never seen companies putting dots in shortnames. Oh wait ...

**Review by nesanton — APPROVED** [review]

**Pull request merged**